File: Task_1_1_2/src/test/java/ru/nsu/smolin/DeckTest.java
package ru.nsu.smolin;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.HashSet;
import java.util.Set;
import org.junit.jupiter.api.Test;
import ru.nsu.smolin.model.Card;
import ru.nsu.smolin.model.Deck;


/**
 * Tests for Deck class.
 */
final class DeckTest {

    @Test
    void newDeckHas52Cards() {
        Deck d = Deck.create();
        assertEquals(52, d.size());
    }

    @Test
    void newDeckContainsOnlyUniqueCards() {
        Deck d = Deck.create();
        Set<Card> set = new HashSet<>();
        for (int i = 0; i < d.size(); i++) {
            set.add(d.get(i));
        }
        assertEquals(52, set.size());
    }

    @Test
    void drawReducesDeckSize() {
        Deck d = Deck.create();
        d.draw();
        assertEquals(51, d.size());
    }
}

File: Task_1_1_2_Kotlin/src/test/kotlin/ru/nsu/smolin/DeckTest.kt
package ru.nsu.smolin

import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class DeckTest {
    @Test
    fun `new deck init with 52 card`() {
        val d = Deck()
        assertEquals(52, d.size())
    }

    @Test
    fun `new deck contains only unique cards`() {
        val d = Deck()
        val set = d.toSet()
        assertEquals(52, set.size)
    }

    @Test
    fun `check draw`() {
        val d = Deck()
        d.draw()
        assertEquals(51, d.size())
    }
}